cpp/dolfinx/io/utils.h

```diff
@@ -0,0 +1,499 @@
+// Copyright (C) 2012-2024 Chris N. Richardson, Jørgen S. Dokken
+//
+// This file is part of DOLFINx (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <array>
+#include <basix/mdspan.hpp>
+#include <dolfinx/common/types.h>
+#include <span>
+#include <utility>
+#include <vector>
+
+namespace
```

Comment:
Anonymous namespaces shouldn't be used in header files. Move the typedef into the function body.